connectivity/FEATURE_BLE/source/cordio/source/GattServerImpl.cpp

```diff
@@ -1146,8 +1148,23 @@ uint8_t GattServer::atts_read_cb(
             return read_auth_params.authorizationReply & 0xFF;
         }
 
-        pAttr->pValue = read_auth_params.data;
-        *pAttr->pLen = read_auth_params.len;
+        /* if new data provided copy into the attribute value buffer */
+        if (read_auth_params.data) {
+            if (read_auth_params.len > pAttr->maxLen) {
+                tr_error("Read authorisation callback set length larger than maximum attribute length, "
+                         "cannot copy data");
+                err = ATT_ERR_UNLIKELY;
```

Comment:
```suggestion
                err = ATT_ERR_LENGTH;
```

Reply:
ATT_ERR_LENGTH has a specific meaning and this is not it. This error will be sent to the remote device.